skins/oasis/modules/UserPagesHeaderController.class.php

```diff
@@ -71,325 +74,166 @@ public static function getUserName( Title $title, $namespaces, $fallbackToGlobal
-			'selected' => ( $wgTitle->isSpecial( 'Contributions' ) ),
+		$tabs[] = [
+			'link' => Linker::linkKnown( SpecialPage::getTitleFor( "Contributions/{$userName}" ), wfMessage( 'contris_s' )->escaped() ),
+			'selected' => ( $this->wg->Title->isSpecial( 'Contributions' ) ),
 			'data-id' => 'contribs',
-		);
+		];
 
 		if ( self::isItMe( $userName ) ) {
 			// following (only render when user is viewing his own user pages)
-			if ( !empty( $wgEnableWikiaFollowedPages ) ) {
-				$tabs[] = array(
-					'link' => Wikia::link( SpecialPage::getTitleFor( 'Following' ), wfMsg( 'wikiafollowedpages-following' ) ),
-					'selected' => ( $wgTitle->isSpecial( 'Following' ) ),
+			if ( !empty( $this->wg->EnableWikiaFollowedPages ) ) {
+				$tabs[] = [
+					'link' => Linker::linkKnown( SpecialPage::getTitleFor( 'Following' ), wfMessage( 'wikiafollowedpages-following' )->escaped() ),
+					'selected' => ( $this->wg->Title->isSpecial( 'Following' ) ),
 					'data-id' => 'following',
-				);
+				];
 			}
-			if ( !empty( F::app()->wg->EnableUserActivityExt ) ) {
+			if ( !empty( $this->wg->EnableUserActivityExt ) ) {
 				$tabs[] = [
-					'link' => Wikia::link( SpecialPage::getTitleFor( 'UserActivity' ), wfMessage( 'user-activity-tab' )->text() ),
-					'selected' => ( $wgTitle->isSpecial( 'UserActivity' ) ),
+					'link' => Linker::linkKnown( SpecialPage::getTitleFor( 'UserActivity' ), wfMessage( 'user-activity-tab' )->escaped() ),
+					'selected' => ( $this->wg->Title->isSpecial( 'UserActivity' ) ),
 					'data-id' => 'user-activity',
 				];
 			}
 
 			// avatar dropdown menu
-			$this->avatarMenu = array(
-				Wikia::link( SpecialPage::getTitleFor( 'Preferences' ), wfMsg( 'oasis-user-page-change-avatar' ) )
-			);
+			$this->avatarMenu = [
+				Linker::linkKnown( SpecialPage::getTitleFor( 'Preferences' ), wfMessage( 'oasis-user-page-change-avatar' )->escaped() ),
+			];
 		}
 
-		wfRunHooks( 'UserPagesHeaderModuleAfterGetTabs', array( &$tabs, $namespace, $userName ) );
+		wfRunHooks( 'UserPagesHeaderModuleAfterGetTabs', [ &$tabs, $namespace, $userName ] );
 
 		wfProfileOut( __METHOD__ );
 		return $tabs;
 	}
 
-	/**
-	 * Get and format stats for given user
-	 */
-	private function getStats( $userName ) {
-		wfProfileIn( __METHOD__ );
-		global $wgLang;
-
-		$user = User::newFromName( $userName );
-		$stats = array();
-
-		if ( !empty( $user ) && $user->isLoggedIn() ) {
-			$userStatsService = new UserStatsService( $user->getId() );
-			$stats = $userStatsService->getStats();
-
-			if ( !empty( $stats ) ) {
-				// date and points formatting
-				if ( !empty( $stats['firstContributionTimestamp'] ) ) {
-					$stats['date'] = $wgLang->date( wfTimestamp( TS_MW, $stats['firstContributionTimestamp'] ) );
-				}
-				$stats['edits'] = $wgLang->formatNum( $stats['editcount'] );
-			}
-		}
+	public function index() {
+		$userName = self::getUserName( $this->wg->Title, BodyController::getUserPagesNamespaces() );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)